File: Atcoder_#322/ruby/b_prefix_and_suffix.rb
n, m = gets.split.map(&:to_i)
s = gets.chomp
t = gets.chomp

is_prefix = (s == t[0...n])

is_suffix = (s == t[-n..-1])

if is_prefix && is_suffix
  puts 0
elsif is_prefix
  puts 1
elsif is_suffix
  puts 2
else
  puts 3
end

File: Atcoder_#322/javascript/b_prefix_and_suffix.js
function main(input) {
  const lines = input.split('\n');
  const [n, m] = lines[0].split(' ').map(Number);
  const s = lines[1];
  const t = lines[2];

  const isPrefix = (s === t.slice(0, n));

  const isSuffix = (s === t.slice(-n));

  if (isPrefix && isSuffix) {
    console.log(0);
  } else if (isPrefix) {
    console.log(1);
  } else if (isSuffix) {
    console.log(2);
  } else {
    console.log(3);
  }
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));
